cloudinit/config/cc_grub_dpkg.py

```diff
@@ -130,35 +151,55 @@ def handle(
         log.debug("%s disabled by config grub_dpkg/enabled=%s", name, enabled)
         return
 
-    idevs = util.get_cfg_option_str(mycfg, "grub-pc/install_devices", None)
-    if idevs is None:
-        idevs = fetch_idevs(log)
+    dconf_sel = get_debconf_config(mycfg, log)
+    if dconf_sel is None:
+        log.debug(
+            "%s no debconf config returned",
+            name,
+        )
+        return
 
-    idevs_empty = mycfg.get("grub-pc/install_devices_empty")
-    if idevs_empty is None:
-        idevs_empty = not idevs
-    elif not isinstance(idevs_empty, bool):
-        idevs_empty = util.translate_bool(idevs_empty)
-    idevs_empty = str(idevs_empty).lower()
+    log.debug("Setting grub debconf-set-selections with '%s'" % dconf_sel)
 
-    # now idevs and idevs_empty are set to determined values
-    # or, those set by user
+    try:
+        subp.subp(["debconf-set-selections"], dconf_sel)
+    except Exception as e:
+        util.logexc(
+            log, "Failed to run debconf-set-selections for grub-dpkg: %s", e
```

Comment:
Since grub-dpkg is deprecated in favor of grub_dpkg, can we change this error to reflect either the preferred key, or whichever was actually passed?